servo/runner.py

```diff
@@ -223,6 +224,8 @@ async def main_loop(self) -> None:
     def run_main_loop(self) -> None:
         if self._main_loop_task:
             self._main_loop_task.cancel()
+            loop = asyncio.get_event_loop()
+            loop.create_task(self.servo.dispatch_event(servo.Events.startup))
```

Comment:
```suggestion
    async def run_main_loop(self) -> None:
        if self._main_loop_task:
            self._main_loop_task.cancel()
            
        await self.servo.startup()
        self.logger.info(
            f"Servo started with {len(self.servo.connectors)} active connectors [{self.optimizer.id} @ {self.optimizer.url or self.optimizer.base_url}]"
        )
```

I agree we should update the startup/shutdown lifecycle given the implementation of cancel responses. However, we need to update a few more places for the sake of completeness:

* We're moving this startup into `run_main_loop()` (delete the old one): https://github.com/opsani/servox/blob/6ce0cbc39fd003290d0f54a9dcafb19278f399da/servo/runner.py#L257
* await run_main_loop now that its async:
  * https://github.com/opsani/servox/blob/6ce0cbc39fd003290d0f54a9dcafb19278f399da/servo/runner.py#L299
  * https://github.com/opsani/servox/blob/6ce0cbc39fd003290d0f54a9dcafb19278f399da/servo/runner.py#L404
* shutdown the servo during cancellation handling: 
  * https://github.com/opsani/servox/blob/6ce0cbc39fd003290d0f54a9dcafb19278f399da/servo/runner.py#L373-L404
  * https://github.com/opsani/servox/blob/6ce0cbc39fd003290d0f54a9dcafb19278f399da/servo/runner.py#L518-L527